### ISSUE TITLE
Expand scrolling area with scroll wheel of Save/Load list

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#2420]
Currently, we can scroll only if cursor is over list itself or over a scrollbar.
![image](https://user-images.githubusercontent.com/55348946/102484644-9c26b380-4077-11eb-9344-a2760ffb2803.png)
![image](https://user-images.githubusercontent.com/55348946/102484656-a2b52b00-4077-11eb-94d5-6864366d6d6c.png)
Over the other areas scrolling using mouse scroll wheel doesn't work. It would be nice if we can use scroll wheel over any place here and savelist during game.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> Over the other areas scrolling using mouse scroll wheel doesn't work. It would be nice if we can use scroll wheel over any place here and savelist during game.

For information, I made a proposal related to these lists here (https://github.com/ihhub/fheroes2/issues/2118), and it won't be possible to have both except to separate the screen in two vertical zones (one zone for the list and another zone for the selected name). :)

**Comment by Branikolog** [comment]
@LeHerosInconnu 
Oh, completely forgot about this thread!
Do you think, we should be able to use scroll wheel only above this three elements? I suggest game allow using wheel everywhere, as sometimes, while scrolling mouse moves away a bit and scrolling stops. What is your opinion?

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> @LeHerosInconnu
> Oh, completely forgot about this thread!
> Do you think, we should be able to use scroll wheel only above this three elements? I suggest game allow using wheel everywhere, as sometimes, while scrolling mouse moves away a bit and scrolling stops. What is your opinion?

The areas can be extended a little around the elements (or at the whole screen eventually, as I indicated):

![Extended area fh2 01](https://user-images.githubusercontent.com/43583994/102490752-8ca36c80-406f-11eb-97fe-9fa8a8049cf1.png)

![Extended area fh2 02](https://user-images.githubusercontent.com/43583994/102490762-8f9e5d00-406f-11eb-8910-5c087c0a2b62.png)

**Comment by Branikolog** [comment]
Ah, I haven't catch an idea of your issue from the first sight. 
Your proposal at #2118 was about changing of the selected by a yellow color item, using a wheel, while pointing the area under the full list of files. This suggestion is not so comfortable, in my point of view, as while using a mouse wheel to scroll the full list it is so easy to choose a needed file appeared there. Remembering me scrolling this list, I always put cursor any place, it can be even a top or a bottom part of a screen, and dividing screen into two parts would always confuse me...
My topic is about making "scrollable" the whole screen, as if don't point list or a scrollbar - scrolling with wheel doesn't work. Especially often i stop cursor over an are between and scrolling doesn't happen at all with "no reason". :)

**Comment by LeHerosInconnu** [comment]
@Branikolog.

In this case, it would be wiser to extend the area to the whole screen for my proposal (https://github.com/ihhub/fheroes2/issues/2118); there would not even be a need to make a selection in the list, it would be automatic as it is already the case with up and down arrow keys on the keyboard. :)

**Comment by Branikolog** [comment]
@LeHerosInconnu 
So, update, please, your suggestion with this new type scrolling behaviour, And we'll debate there. :)
Regarding my issue, we should make the whole screen active for scrolling. Not only list itself and a small thin scroll bar to the right - but the whole screen.  That's the point.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> @LeHerosInconnu
> So, update, please, your suggestion with this new type scrolling behaviour, And we'll debate there. :)

I have updated my issue https://github.com/ihhub/fheroes2/issues/2118.

> Regarding my issue, we should make the whole screen active for scrolling. Not only list itself and a small thin scroll bar to the right - but the whole screen. That's the point.

Have you tried using up and down arrow keys on the keyboard?
What I meant was that with my updated https://github.com/ihhub/fheroes2/issues/2118 proposal, your https://github.com/ihhub/fheroes2/issues/2420 proposal is no longer necessary. :)

**Comment by Branikolog** [comment]
While using up/down arrows we change the selected name (a name with yellow color) in a list, while using a scroll wheel we simply move entire list. If I understood correctly, you proposed to use the same for wheel scrolling. Am I right? You're talking about selecting file with a wheel, and I'm talking about wheel area.
But for me scrolling, as it works now is more comfortable. I simply want it to work everywhere on the screen.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> While using up/down arrows we change the selected name (a name with yellow color) in a list, while using a scroll wheel we simply move entire list. If I understood correctly, you proposed to use the same for wheel scrolling. Am I right? You're talking about selecting file with a wheel, and I'm talking about wheel area.
> But for me scrolling, as it works now is more comfortable. I simply want it to work everywhere on the screen.

Up and down arrow keys on the keyboard do both at the same time, they change the selected name **and** move entire list.

I suggest that the mouse wheel does the same on the whole screen (wherever the mouse cursor is positioned on the screen).

**Comment by Branikolog** [comment]
Up and down keys are moving the selected name. And list stays still till the time the selected name moves out of a list. List changes to the state, where selected name is in top/bottom position. While scrolling with mouse moves entire list only on 1 name lower each click.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> Up and down keys are moving the selected name. And list stays still till the time the selected name moves out of a list. List changes to the state, where selected name is in top/bottom position. While scrolling with mouse moves entire list only on 1 name lower each click.

Is this really a problem?

**Comment by Branikolog** [comment]
@LeHerosInconnu.
No, it is not. :) It's just a different issues. Excuse me, if I understand something wrong... In your issue you're talking about the type of scrolling, where the selection of names proceeds with the mouse.
And I've create this issue, as I cannot scroll the list (No matter, how does that scrolling work) any place, I want. But only, while pointing a certain elements of UI, which is not ok.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> @LeHerosInconnu.
> No, it is not. :) It's just a different issues. Excuse me, if I understand something wrong... In your issue you're talking about the type of scrolling, where the selection of names proceeds with the mouse.
> And I've create this issue, as I cannot scroll the list (No matter, how does that scrolling work) any place, I want. But only, while pointing a certain elements of UI, which is not ok.

Your issue will be resolved at the same time (the list will scroll as the selected file moves), and it will not be necessary to select the file name from the list by clicking on it with the mouse (although it will still be possible), it will be done automatically. :)

**Comment by Branikolog** [comment]
I have a counter argument to your proposal. :)
When we enter save/load list, for example, we always watch the top name being selected. So, we need to make about 1 full wheel scroll to make the lowest name in visible part of become yellow, After that, our list start moving down. So we need two moves of our finger. Agreed, when the needed name lies not far from the current, it's quite comfortable to make a smal touch if wheel and select it. But when we need to scroll much more list it's not so simple.
With current scrolling, we can make even short, single click to show more files from a hidden part of the a list.
That goes without saying, that current keyboard scrolling with arrow keys I find rather uncomfortable, as list jumps too much, at the time the selected name reaches edge of visible part of list. After whole visible part of list changes, it's rather hard to find needed name. 
A small example: lets say, that I'm looking a name, which is located visible list -2 name. I start scrolling down, and when the selected filename reaches bottom, list completely changes. But logically I expect new file appear in the bottom of the list. So I have to look up and search needed file there. I understand, I've been accustomed to the scrolling manner... But I feel like that. Nothing can do.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> When we enter save/load list, for example, we always watch the top name being selected. So, we need to make about 1 full wheel scroll to make the lowest name in visible part of become yellow,

No, if the name is already visible in the list, just click on it to select it.

> With current scrolling, we can make even short, single click to show more files from a hidden part of the a list.

This will still be possible.

> A small example: lets say, that I'm looking a name, which is located visible list -2 name. I start scrolling down, and when the selected filename reaches bottom, list completely changes.

If it's the list that "jumps" the problem, it is to be done in such a way that the list does not jump, neither with the mouse wheel, nor with the up and down keys. :)

**Comment by Branikolog** [comment]
We have to decide whether list should jump or not. Currently, scrolling with mouse works smoothly, and with keyboard it jumps. 
Anyway, I think two current issues should be separate, as we're talking about different problems. 
This issue is about inability to use mouse scroll, and yours is about the logic of scrolling (jumping or smooth scrolling).

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> We have to decide whether list should jump or not. Currently, scrolling with mouse works smoothly, and with keyboard it jumps.

I've always found these lists that "jump" a bit awkward.
For me, it would be preferable that the lists scroll smoothly in all cases, rather than jumping as it does now with the up and down keys.

> Anyway, I think two current issues should be separate, as we're talking about different problems.
> This issue is about inability to use mouse scroll, and yours is about the logic of scrolling (moving the "yellow" selection down or not).

Yes, you can create a new issue to make the lists scroll smoothly in all cases (with mouse wheel and with up and down keys).

**Comment by Branikolog** [comment]
> No, if the name is already visible in the list, just click on it to select it.

I mean other thing: when we need to reach save file, which is out of visible list, we have to scroll the whole visible list first (as the first selected file is always on top) and only after we can reach needed file. It's in the case where mouse affects the selected yellow name. 
Current scrolling allows us to shift a list at the moment, we touch a wheel. And then choose needed file by left click. 
Both this procedures have similar effectiveness, but I personally prefer scrolling the whole list and choose file with left click, rather than scrolling with wheel, till I reach the file and then press OK.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> > No, if the name is already visible in the list, just click on it to select it.
> 
> I mean other thing: when we need to reach save file, which is out of visible list, we have to scroll the whole visible list first (as the first selected file is always on top) and only after we can reach needed file. It's in the case where mouse affects the selected yellow name.
> Current scrolling allows us to shift a list at the moment, we touch a wheel. And then choose needed file by left click.
> Both this procedures have similar effectiveness, but I personally prefer scrolling the whole list and choose file with left click, rather than scrolling with wheel, till I reach the file and then press OK.

@ihhub requested me to stop posting suggestions for now.
He wanted to get clear about the remaining issues; it was without counting on bugs. :D

So I have a lot of suggestions pending.
One of them is to make the scrollbar buttons allow for continuous scrolling of the lists (or that the buttons allow to increase/decrease the number of creatures continuously, for example in other windows, etc.) when the user makes a held left-click on the said buttons (instead of having to make a lot of single left-clicks).
So, by holding down the left mouse button the list scrolls continuously, which is not the case at the moment (one left click the list moves a bit, a second left click the list moves a bit, etc.).

**Comment by ihhub** [comment]
My apologies for interrupting such a delightful conversation but @LeHerosInconnu is right about my request about improvements. We still have 500+ bugs so creating more items will make the work of developers endless.

Why don't we take such conversations to another place and once we come up with something good we can create an issue in GitHub. Every person could have his own opinion but we need to find the best solution which fits everyone (or the majority of people).

I created a Discussion board within this GitHub project - https://github.com/ihhub/fheroes2/discussions Let's try to use it. If it doesn't work then we come back here, close Discussions and pollute with ideas :+1: 
What do you say, @LeHerosInconnu and @Branikolog ? :)

**Comment by Branikolog** [comment]
@LeHerosInconnu 
Oh, I see. 
I don't have a horde of enhancements like you. So I've decided to make a couple not global. 
Nevertheless, @ihhub has already made me a call down regarding working on enhancements, rather, than bugs. :D
Anyway, this issue is about polishing the current interface features. 
Between, I like your idea of smooth continuous scrolling. ;)
I hope, in future our team will implement it.)

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> My apologies for interrupting such a delightful conversation but @LeHerosInconnu is right about my request about improvements. We still have 500+ bugs so creating more items will make the work of developers endless.
> 
> Why don't we take such conversations to another place and once we come up with something good we can create an issue in GitHub. Every person could have his own opinion but we need to find the best solution which fits everyone (or the majority of people).
> 
> I created a Discussion board within this GitHub project - https://github.com/ihhub/fheroes2/discussions Let's try to use it. If it doesn't work then we come back here, close Discussions and pollute with ideas 
> What do you say, @LeHerosInconnu and @Branikolog ? :)

It's okay for me. :)

**Comment by Branikolog** [comment]
Agreed.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

> At max what we can do is to cover the full area for the list, scrollbar and buttons. This is the way how almost every modern application works. We shouldn't allow scrolling if the mouse is out of the area.

Covering the entire window surface for mouse wheel use would be a plus.
However, as there is only this one window on the screen, it would not be a problem to extend the use of the mouse wheel to the entire screen surface.
As a result, the user can scroll the list without the mouse cursor being located inside the window, which is more comfortable to use.
For example, the user can start scrolling the list even if the mouse cursor is located outside the window, while he moves the mouse cursor inside the window to make his selection.
